keps/sig-node/4540-strict-cpu-reservation/README.md

```diff
@@ -0,0 +1,527 @@
+  - Will enabling / disabling the feature require downtime of the control plane? No
+  - Will enabling / disabling the feature require downtime or reprovisioning of a node?  Yes -- removing /var/lib/kubelet/cpu\_manager\_state and restarting kubelet are required.
+
+
+###### Does enabling the feature change any default behavior?
+
+Yes. Reserved CPU cores will be strictly used for system daemons and interrupt processing no longer available for workloads.
+
+The feature is only enabled when all following conditions are met:
+1. The `CPUManagerPolicyAlphaOptions` feature gate must be enabled
+2. The `static` `CPUManager` policy must be selected
+3. The new `strict-cpu-reservation` policy option must be selected
+4. The `reservedSystemCPUs` is not empty
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes, the feature can be disabled by either:
+1. Disabling the `CPUManagerPolicyAlphaOptions` feature gate
+2. Removing `strict-cpu-reservation` from the list of `CPUManager` policy options
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The feature will be enabled regardless it is enabled for the first time or not.
+
+###### Are there any tests for feature enablement/disablement?
+
+- A specific e2e test will demonstrate that the default behaviour is preserved when the feature gate is disabled, or when the feature is not used (2 separate tests)
+
+### Rollout, Upgrade and Rollback Planning
+
+<!--
+This section must be completed when targeting beta to a release.
+-->
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+If the feature rollout fails, burstable and best-efforts can run on the reserved CPU cores.
+If the feature rollback fails, burstable and best-efforts can not run on the reserved CPU cores.
+In either case, existing workload will not be affected.
+
+<!--
+Try to be as paranoid as possible - e.g., what if some components will restart
+mid-rollout?
+
+Be sure to consider highly-available clusters, where, for example,
+feature flags will be enabled on some API servers and not others during the
+rollout. Similarly, consider large clusters and how enablement/disablement
+will rollout across nodes.
+-->
+
+###### What specific metrics should inform a rollback?
+
+<!--
+What signals should users be paying attention to when the feature is young
+that might indicate a serious problem?
+-->
+
+When the feature rolls back, the reserved CPU cores are included in the `defaultCpuSet` in /var/lib/kubelet/cpu\_manager\_state file.
+
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+<!--
+Describe manual testing that was done and the outcomes.
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+We manually test it in our internal environment and it works.
```

Comment:
For beta promotion you'll need to be much more descriptive, what the manual testing looked like. Not needed now.

Reply:
got it